src/Drips.sol

```diff
@@ -251,10 +257,23 @@ library Drips {
             newReceivers,
             newDefaultEnd
         );
-        state.dripsHash = hashDrips(newReceivers);
         state.updateTime = _currTimestamp();
         state.balance = newBalance;
-        emit DripsUpdated(userId, assetId, newBalance, newReceivers);
+        bytes32 newDripsHash = hashDrips(newReceivers);
+        if (newDripsHash != currDripsHash) {
+            state.dripsHash = newDripsHash;
+            for (uint256 i = 0; i < newReceivers.length; i++) {
+                DripsReceiver memory receiver = newReceivers[i];
+                emit DripsReceiverSeen(
+                    newDripsHash,
+                    receiver.userId,
+                    receiver.amtPerSec,
+                    receiver.start,
+                    receiver.duration
+                );
```

Comment:
Because it would be quite cumbersome to do there, this function already does a lot, has a complex flow and can barely handle all its local variables. If the cost of the additional iteration isn't too high, I'd prefer keeping it in a separate loop and close to the hash updating logic for the sake of clarity.

Reply:
It is always the trade-off between gas costs and code complexity. I remember that it is not that expensive to iterate. Let's keep it for now.